Shopify/cli
# Add authCallbackPath to web.toml configuration

### WHY are these changes introduced?

If a developer chooses any other auth callback path other than the three generated by the CLI, they'll have to update via the partner dashboard manually.

Fixes https://github.com/Shopify/first-party-library-planning/issues/400

### WHAT is this pull request doing?

Adds a `authCallbackPath` configuration to the `shopify.web.toml` file that is added to the generated Partner Dashboard URLs

### How to test your changes?

1. In an existing app, modify the `web/shopify.web.toml` file to include a `authCallbackPath` value, e.g.,
   ```ini
   authCallbackPath="/my/new/auth/callback/path"
   ```
2. Modify the app so that the callback path used in the app matches that in the configuration file.
3. Run the cli against an existing app
   ```shell
   dev shopify app dev --path /directory/path/to-a/shopify/app
   ```
4. Check the callback URLs in the partner dashboard to make sure it includes the path above.

### Post-release steps

TBD

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
